#! /usr/bin/env python

PACKAGE='htp_auto'

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()
#       Name       Type      Level Description     Default Min   Max
gen.add("travel_vel", double_t,    0,    "Forward velocity between waypoints (m/s)", .2, 0, 1.0)

gen.add("heading_kp", double_t,    0,    "Propertional gain.",   5,  0,  100)
gen.add("heading_ki", double_t,    0,    "Integral gain.",       0,  0,  100)
gen.add("heading_kd", double_t,    0,    "Derivative gain.",     0,  0,  100)

gen.add("heading_sat_high", double_t,  0,  "Maximum angular velocity (rad/sec).",  1.0,  0,  6)
gen.add("heading_sat_low",  double_t,  0,  "Minimum angular velocity (rad/sec).",  -1.0,  -6,  0)

gen.add("heading_int_sat_high", double_t,  0,  "Maximum integral sum (rad-sec).",  0.5,  0,  6)
gen.add("heading_int_sat_low",  double_t,  0,  "Minimum integral sum (rad-sec).",  -0.5,  -6,  0)

gen.add("heading_pub_prescaler", int_t, 0,  "Scales down publish rate. -1 to disable.", 5, -1, 100)

exit(gen.generate(PACKAGE, "waypoint_guidance", "GuidanceParams"))
